compiler/rustc_codegen_llvm/src/gotoc/utils.rs

```diff
@@ -133,7 +138,9 @@ impl Type {
     /// Checks if the struct represents a Rust "Unique"
     pub fn is_rust_unique_pointer(&self) -> bool {
         self.type_name().map_or(false, |name| {
-            name.starts_with("tag-std::ptr::Unique") || name.starts_with("tag-core::ptr::Unique")
+            name.starts_with("tag-std::ptr::Unique")
+                || name.starts_with("tag-core::ptr::Unique")
+                || name.starts_with("tag-rustc_std_workspace_core::ptr::Unique")
```

Comment:
Why not use `name.ends_with("::ptr::Unique")` here? It seems you are trying to match the full name anyway.

Reply:
The name then ends with some monomorphized <T> etc